src/Compilers/CSharp/Test/Symbol/Symbols/FunctionPointerTypeSymbolTests.cs

```diff
@@ -268,36 +282,61 @@ class C
+                Diagnostic(ErrorCode.ERR_InvalidFunctionPointerCallingConvention, "").WithArguments("").WithLocation(6, 47));
 
             var c = comp.GetTypeByMetadataName("C");
-            var m = c.GetMethod("M1");
-            var pointerType = (FunctionPointerTypeSymbol)m.Parameters.Single().Type;
-            FunctionPointerUtilities.CommonVerifyFunctionPointer(pointerType);
-            Assert.Equal(CallingConvention.Default, pointerType.Signature.CallingConvention);
+            var m1 = c.GetMethod("M1");
+            var m1PointerType = (FunctionPointerTypeSymbol)m1.Parameters.Single().Type;
+            Assert.Equal(CallingConvention.Unmanaged, m1PointerType.Signature.CallingConvention);
 
-            // PROTOTYPE(func-ptr): Verify the public results of M2 and M3
+            var m2 = c.GetMethod("M2");
+            var m2PointerType = (FunctionPointerTypeSymbol)m2.Parameters.Single().Type;
+            Assert.Equal(CallingConvention.Unmanaged, m2PointerType.Signature.CallingConvention);
+
+            var m3 = c.GetMethod("M3");
+            var m3PointerType = (FunctionPointerTypeSymbol)m3.Parameters.Single().Type;
+            Assert.Equal(CallingConvention.Unmanaged, m3PointerType.Signature.CallingConvention);
 
             // https://github.com/dotnet/roslyn/issues/43321 test public calling convention exposure when added to the API
             var syntaxTree = comp.SyntaxTrees[0];
             var model = comp.GetSemanticModel(syntaxTree);
+            var functionPointers = syntaxTree.GetRoot().DescendantNodes().OfType<FunctionPointerTypeSyntax>().ToArray();
+            Assert.Equal(3, functionPointers.Length);
 
-            FunctionPointerUtilities.VerifyFunctionPointerSemanticInfo(model,
-                syntaxTree.GetRoot().DescendantNodes().OfType<FunctionPointerTypeSyntax>().First(),
+            FunctionPointerUtilities.VerifyFunctionPointerSemanticInfo(model, functionPointers[0],
                 expectedSyntax: "delegate* unmanaged[invalid]<void>",
-                expectedType: "delegate*<System.Void>",
-                expectedSymbol: "delegate*<System.Void>");
+                expectedType: "delegate*<System.Void modopt(CallConvinvalid)>",
+                expectedSymbol: "delegate*<System.Void modopt(CallConvinvalid)>");
+
+            FunctionPointerUtilities.VerifyFunctionPointerSemanticInfo(model, functionPointers[1],
+                expectedSyntax: "delegate* unmanaged[invalid, Stdcall]<void>",
+                expectedType: "delegate*<System.Void modopt(CallConvinvalid) modopt(System.Runtime.CompilerServices.CallConvStdcall)>",
+                expectedSymbol: "delegate*<System.Void modopt(CallConvinvalid) modopt(System.Runtime.CompilerServices.CallConvStdcall)>");
+
+            FunctionPointerUtilities.VerifyFunctionPointerSemanticInfo(model, functionPointers[2],
+                expectedSyntax: "delegate* unmanaged[]<void>",
+                expectedType: "delegate*<System.Void modopt(CallConv)>",
```

Comment:
>modopt(CallConv) [](start = 53, length = 16)

This doesn't look right. We probably shouldn't add any modifiers for this syntax error case.